src/geo/mercator_coordinate.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+
+import LngLat from '../geo/lng_lat';
+import type {LngLatLike} from '../geo/lng_lat';
+
+/*
+ * The circumference of the world in meters at the given latitude.
+ */
+function circumferenceAtLatitude(latitude: number) {
+    const circumference = 2 * Math.PI * 6378137;
+    return circumference * Math.cos(latitude * Math.PI / 180);
+}
+
+/**
+ * A `MercatorCoordinate` object represents a 3 dimensional position in projected web mercator coordinates.
+ *
+ * The "world size" used by `MercatorCoordinate` is 1, meaning `MercatorCoordinate(0, 0, 0)` is the north-west
+ * corner of the mercator world and `MercatorCoordinate(1, 1, 0)` is the south-east corner of the mercator world.
```

Comment:
This documentation provides great clarity, I really appreciate it.

A lot of geospatial people might be confused by this definition though since the [web mercator projection](https://epsg.io/3857) is in meters and has a projected bounds of

```
-20026376.39 -20048966.10
20026376.39 20048966.10
```

Since `MertactorCoordinate` here is a a unit vector, I'm wondering if we need to be more explict that this is not ESPG:3857 web mercator :thinking: ?

Reply:
@andrewharvey thats for flagging this! I'll have to think more closely whether we should match that or if we should just clarify the documentation.

Reply:
I clarified this a bit in 83cda37. Do you think this is clear enough?

*Also, I think the projected bounds on that website are wrong. I think it should be `20037508`.*

> I like the simplicity of a unit vector as the underlying representation compared to EPSG:3857, however I very much see it as an internal implementation detail.

Yep, it makes conversion from vector tiles to these units and back simpler. It also allows vector tile positions to be stored without loss of precision. I think it makes sense to keep it as is.

Do you think renaming it would help? `UnitMercatorCoordinate` to emphasize the unitness of it? or `TiledMercatorCoordinate` to nod in the direction of the tiled coordinate spaces this borrows from? I think I'm leaning towards just `MercatorCoordinate`

Reply:
I like the new documentation in https://github.com/mapbox/mapbox-gl-js/commit/83cda371cd4369f5caab946823bb4aff67a56f4e :+1: 

-1 to `TiledMercatorCoordinate`, that makes me think xyz tiles which vary by zoom level, but in fact this `MercatorCoordinate` doesn't vary my zoom level and isn't broken into tiles.

I like `UnitMercatorCoordinate` since it's more explicit that these are unit coordinates (0 to 1),